tests/test_web_response.py

```diff
@@ -1090,6 +1090,31 @@ def test_response_with_immutable_headers() -> None:
                             'Content-Type': 'text/plain; charset=utf-8'}
 
 
+async def test_response_prepared_after_header_preparation() -> None:
+    req = make_request('GET', '/')
+    resp = StreamResponse()
+    await resp.prepare(req)
+
+    assert type(resp.headers['Server']) is str
```

Comment:
I think we should also take inheritance into account:
```suggestion
    assert isinstance(resp.headers['Server'], str)
```
WDYT?

Reply:
No, the strict type check is better here.
Please keep the test as is